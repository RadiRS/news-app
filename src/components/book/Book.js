import React from 'react';
import styled from 'styled-components/native';
import { Images, Fonts } from '../../themes';
import { formatTime } from '../../helpers';

const Book = props => {
  const { multimedia = [], book_details, published_date } = props.book;

  return (
    <Container>
      <Content>
        <Title numberOfLines={2}>{book_details[0].title}</Title>
        <Author>{book_details[0].contributor}</Author>
        <Date>{formatTime(published_date)}</Date>
      </Content>
      <Cover>
        {multimedia.length > 1 ? (
          <Image
            source={{ uri: `https://static01.nyt.com/${multimedia[3].url}` }}
          />
        ) : (
          <Image source={{ uri: Images.noImage }} />
        )}
      </Cover>
    </Container>
  );
};

export default Book;

const Container = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: white;
  height: 150px;
  border-radius: 14px;
  margin: 10px 20px 10px 20px;
  elevation: 10;
`;

const Content = styled.View`
  flex: 0.7;
  padding-left: 10px;
`;

const Cover = styled.View`
  flex: 0.3;
  align-items: center;
  justify-content: center;
`;

const Image = styled.Image`
  width: 100px;
  height: 100px;
  border-radius: 14px;
`;

const Title = styled.Text`
  font-size: 22px;
  font-weight: bold;
`;

const Author = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;

const Date = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;
