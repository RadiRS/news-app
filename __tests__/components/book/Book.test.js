import 'react-native';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import Book from '../../../src/components/book/Book';

test('renders Book component correctly', () => {
  const tree = renderer.create(<Book book={book} />).toJSON();
  expect(tree).toMatchSnapshot();
});

const book = {
  multimedia: [],
  book_details: [{ title: '', contributor: '' }],
  published_date: ''
};
