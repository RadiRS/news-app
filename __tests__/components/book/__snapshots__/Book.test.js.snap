// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Book component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 14,
        "elevation": 10,
        "flexDirection": "row",
        "height": 150,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 0.3,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://lorempixel.com/640/480",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 14,
            "height": 100,
            "width": 100,
          },
        ]
      }
    />
  </View>
</View>
`;
